src/integrationTest/java/com/mongodb/kafka/connect/avro/SchemaNameIntegrationTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.avro;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+
+import io.confluent.connect.avro.AvroData;
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.avro.AvroSchema;
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import com.mongodb.kafka.connect.mongodb.MongoKafkaTestCase;
+
+final class SchemaNameIntegrationTest extends MongoKafkaTestCase {
```

Comment:
I think we can do this with a unit test, saving spinning up the local KAFKA instance. 

Since all we care about it the round trip of the `AvroConverter`, the Schema Registry can be treated as a black box. So this can be simplified.  I adapted your example with the code in [AvroConverterTest](https://github.com/confluentinc/schema-registry/blob/c2b626c1515e1b2ddc8a845b09a00c8a6d7ac820/avro-converter/src/test/java/io/confluent/connect/avro/AvroConverterTest.java#L58) and used the `MockSchemaRegistryClient`. 

eg:

```
  @Test
  public void testAvroSchemaAndValueConversion() {

    final MockSchemaRegistryClient schemaRegistry = new MockSchemaRegistryClient();
    final AvroConverter converter = new AvroConverter(schemaRegistry);
    converter.configure(Collections.singletonMap("schema.registry.url", "http://fake-url"), false);
    String TOPIC = "topic";

    String jsonSchema =
        "{\n"
            + "  \"name\": \"recordSchemaName\",\n"
            + "  \"namespace\": \"recordSchemaNamespace\",\n"
            + "  \"version\": 1,\n"
            + "  \"type\": \"record\",\n"
            + "  \"fields\": [\n"
            + "    {\n"
            + "      \"name\": \"nestedRecordFieldName\",\n"
            + "      \"type\": {\n"
            + "        \"name\": \"nestedRecordSchemaName\",\n"
            + "        \"namespace\": \"nestedRecordSchemaNamespace\",\n"
            + "        \"type\": \"record\",\n"
            + "        \"fields\": [\n"
            + "          {\n"
            + "            \"name\": \"fieldName\",\n"
            + "            \"type\": \"int\"\n"
            + "          }\n"
            + "        ]\n"
            + "      }\n"
            + "    }\n"
            + "  ]\n"
            + "}";

    // Test parsing is as expected
    Schema schema = com.mongodb.kafka.connect.source.schema.AvroSchema.fromJson(jsonSchema);
    SchemaBuilder builder =
            SchemaBuilder.struct()
                    .name("recordSchemaNamespace.recordSchemaName")
                    .field(
                            "nestedRecordFieldName",
                            SchemaBuilder.struct()
                                    .name("nestedRecordSchemaNamespace.nestedRecordSchemaName")
                                    .field("fieldName", SchemaBuilder.INT32_SCHEMA)
                                    .build());
    assertEquals(builder.build(), schema);

    // Create connect schema and value
    Struct value =
        new Struct(schema)
            .put(
                "nestedRecordFieldName",
                new Struct(schema.field("nestedRecordFieldName").schema()).put("fieldName", 1));
    SchemaAndValue schemaAndValue = new SchemaAndValue(schema, value);

    // Test avro data conversion to connect
    AvroSchema avroSchema = AvroUtils.parseSchema(jsonSchema);
    GenericRecord avroRecord =
        new GenericRecordBuilder(avroSchema.rawSchema())
            .set(
                "nestedRecordFieldName",
                new GenericRecordBuilder(
                        avroSchema.rawSchema().getField("nestedRecordFieldName").schema())
                    .set("fieldName", 1)
                    .build())
            .build();

    AvroDataConfig avroDataConfig =
        new AvroDataConfig.Builder()
            .with(AvroDataConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_CONFIG, true)
            .build();
    AvroData avroData = new AvroData(avroDataConfig);
    SchemaAndValue avroSchemaAndValue = avroData.toConnectData(avroSchema.rawSchema(), avroRecord);
    assertEquals(avroSchemaAndValue, schemaAndValue);

    // Test data conversion from connect (where version is added by the registry)
    Schema versionedSchema = builder.version(1).build();
    Struct versionedValue =
            new Struct(versionedSchema)
                    .put(
                            "nestedRecordFieldName",
                            new Struct(versionedSchema.field("nestedRecordFieldName").schema())
                                    .put("fieldName", 1));
    SchemaAndValue versionedSchemaAndValue = new SchemaAndValue(versionedSchema, versionedValue);

    byte[] converted =
            converter.fromConnectData(TOPIC, schemaAndValue.schema(), schemaAndValue.value());
    SchemaAndValue convertedSchemaAndValue = converter.toConnectData(TOPIC, converted);
    assertEquals(versionedSchemaAndValue, convertedSchemaAndValue);
  }
```

Reply:
Thank you, this is great. But since the proposed test uses `AvroConverter`, there is no need to use `AvroData` explicitly anymore: it is used by `AvrConverter`, and I was originally using `AvroData` only to emulate what `AvroConverter` does`.